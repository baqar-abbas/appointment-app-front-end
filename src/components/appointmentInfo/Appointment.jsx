import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addAppointmentThunk } from '../../features/appointmentSlice';
// import { fetchDoctors } from '../../features/doctorSlice';
// import { fetchDoctors } from '../../features/doctorSlice';
const fetchDoctors = {}; // delete this line and uncomment line above

const Appointment = () => {
  const dispatch = useDispatch();

  const [appointmentDate, setAppointmentDate] = useState('');
  const [doctorId, setDoctorId] = useState('');
  // const [patientId, setPatientId] = useState('');
  const [street, setStreet] = useState('');
  const [city, setCity] = useState('');
  const [state, setState] = useState('');
  const [zipCode, setZipCode] = useState('');

  const listDoctors = useSelector((state) => state.doctors.doctors);

  useEffect(() => { // Fetch doctors on component mount
    dispatch(fetchDoctors());
  }, [dispatch]);

  const handleAddAppointment = async (e) => {
    e.preventDefault();

    const appointmentData = {
      appointment: {
        appointment_date: appointmentDate,
        doctor_id: doctorId,
        patient_id: sessionStorage.getItem('userId'),
        status: {
          active: true,
          expire: false,
          cancel: false,
        },
        location: {
          street,
          city,
          state,
          zip_code: zipCode,
        },
      },
    };

    try {
      dispatch(addAppointmentThunk(appointmentData));
      setAppointmentDate('');
      setDoctorId('');
      setStreet('');
      setCity('');
      setState('');
      setZipCode('');

      // Optionally, display a success message to the user.
    } catch (error) {
      error.message = 'Failed to create appointment';
    }
  };

  const handleDoctorSelect = (e) => {
    const selectedDoctorId = parseInt(e.target.value, 10);
    setDoctorId(selectedDoctorId);
  };
  return (
    <div>
      <form className="container mt-4" onSubmit={handleAddAppointment}>
        <div className="form-group row">
          {/* <label htmlFor="appointmentDate" className="col-sm-2 col-form-label">
            Appointment Date:
          </label> */}
          <div className="col-sm-10">
            <input
              type="datetime-local"
              className="form-control"
              id="appointmentDate"
              placeholder="Enter appointment date"
              required
              value={appointmentDate}
              onChange={(e) => setAppointmentDate(e.target.value)}
            />
          </div>
        </div>

        <div className="form-group row">
          {/* <label htmlFor="doctorId" className="col-sm-2 col-form-label">
            Doctor name:
          </label> */}
          <div className="col-sm-10">
            <select
              className="form-control"
              id="doctorId"
              name="doctorId"
              placeholder="Select a doctor"
              required
              value={doctorId}
              onChange={handleDoctorSelect}
            >
              <option value="">Select a doctor</option>
              {listDoctors.map((doctor) => (
                <option key={doctor.id} value={doctor.id}>
                  {doctor.name}
                </option>
              ))}
            </select>
          </div>
        </div>

        <div className="form-group row">
          {/* <label htmlFor="street" className="col-sm-2 col-form-label">
            Street:
          </label> */}
          <div className="col-sm-10">
            <input
              type="text"
              className="form-control"
              placeholder="Street"
              id="street"
              name="street"
              required
              value={street}
              onChange={(e) => setStreet(e.target.value)}
            />
          </div>
        </div>

        <div className="form-group row">
          {/* <label htmlFor="city" className="col-sm-2 col-form-label">
            City:
          </label> */}
          <div className="col-sm-10">
            <input
              type="text"
              className="form-control"
              placeholder="City"
              id="city"
              required
              name="city"
              value={city}
              onChange={(e) => setCity(e.target.value)}
            />
          </div>
        </div>

        <div className="form-group row">
          {/* <label htmlFor="state" className="col-sm-2 col-form-label">
            State:
          </label> */}
          <div className="col-sm-10">
            <input
              type="text"
              required
              className="form-control"
              id="state"
              name="state"
              placeholder="State"
              value={state}
              onChange={(e) => setState(e.target.value)}
            />
          </div>
        </div>

        <div className="form-group row">
          {/* <label htmlFor="zipCode" className="col-sm-2 col-form-label">
            Zip Code:
          </label> */}
          <div className="col-sm-10">
            <input
              type="text"
              className="form-control"
              id="zipCode"
              name="zipCode"
              required
              placeholder="Zip Code"
              value={zipCode}
              onChange={(e) => setZipCode(e.target.value)}
            />
          </div>
        </div>

        <div className="form-group row">
          <div className="col-sm-12 text-center">
            <button type="submit" className="btn btn-primary">
              Add Appointment
            </button>
          </div>
        </div>
      </form>

    </div>
  );
}

export default Appointment;