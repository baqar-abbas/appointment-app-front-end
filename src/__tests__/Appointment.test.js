import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../features/store';
import Appointment from '../components/appointmentInfo/Appointment';
import React from 'react';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Appointment />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});