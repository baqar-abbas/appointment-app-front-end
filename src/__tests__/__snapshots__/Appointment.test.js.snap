// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment Component it renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="appointmentBg row"
  >
    <div
      className="px-0 col-md-2"
    >
      <div
        className="d-flex flex-column border-end h-100 bg-white"
      >
        <div
          className="d-flex align-items-center p-3"
        >
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240.1 4.2c9.8-5.6 21.9-5.6 31.8 0l171.8 98.1L448 104l0 .9 47.9 27.4c12.6 7.2 18.8 22 15.1 36s-16.4 23.8-30.9 23.8H32c-14.5 0-27.2-9.8-30.9-23.8s2.5-28.8 15.1-36L64 104.9V104l4.4-1.6L240.1 4.2zM64 224h64V416h40V224h64V416h48V224h64V416h40V224h64V420.3c.6 .3 1.2 .7 1.8 1.1l48 32c11.7 7.8 17 22.4 12.9 35.9S494.1 512 480 512H32c-14.1 0-26.5-9.2-30.6-22.7s1.1-28.1 12.9-35.9l48-32c.6-.4 1.2-.7 1.8-1.1V224z"
              />
            </svg>
          </span>
          <h1
            className="m-0 ms-2"
          >
            Squa
          </h1>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M333.2 322.8l0 0-133.9-146 0 0L146 118.6c7.8-5.1 37-22.6 78-22.6s70.2 17.4 78 22.6L245.7 180l85.6 93.4 27.4-29.8c16.3-17.7 25.3-40.9 25.3-65V149.1c0-19-5.6-37.5-16.1-53.3L327.8 35.6C312.9 13.4 287.9 0 261.2 0h-76c-25.8 0-50.1 12.5-65.1 33.5L81.9 87C70.3 103.2 64 122.8 64 142.8V164c0 23.2 8.4 45.6 23.6 63.1l56 64.2 0 0 83.3 95.6 0 0 91.8 105.3c10 11.5 26.8 14.3 40 6.8l54.5-31.1c17.8-10.2 21.6-34.3 7.7-49.4l-87.7-95.7zM205.2 410.6l-83.3-95.6L27.1 418.5c-13.9 15.1-10.1 39.2 7.7 49.4l55.1 31.5c13 7.4 29.3 4.9 39.4-6.1l75.9-82.6z"
              />
            </svg>
          </span>
        </div>
        <div
          className="d-flex flex-column flex-grow-1 px-0"
        >
          <button
            className=" p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Add Doctor
          </button>
          <button
            className="p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={true}
            type="button"
          >
            Delete Doctor
          </button>
          <a
            activeClassName="active"
            className="p-3"
            href="/doctors"
            isActive={[Function]}
            onClick={[Function]}
          >
            Doctors
          </a>
          <a
            activeClassName="active"
            className="p-3"
            href="/appointment"
            isActive={[Function]}
            onClick={[Function]}
          >
            Appointment
          </a>
          <a
            activeClassName="active"
            className="p-3"
            href="/my_appointments"
            isActive={[Function]}
            onClick={[Function]}
          >
            My Appointments
          </a>
          <button
            className="p-3 dropdown-item"
            data-rr-ui-dropdown-item=""
            onClick={[Function]}
          >
            Sign Out
          </button>
          <button
            className=" p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={false}
            type="button"
          >
            For Super Admin
          </button>
        </div>
        <div
          className="d-flex align-items-center justify-content-between p-3 social_media"
        >
          <a
            className="nav-link"
            disabled={false}
            href="https://www.facebook.com/your_facebook_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.google.com/your_google_plus_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.linkedin.com/your_linkedin_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.google.com/your_google_drive_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M339 314.9L175.4 32h161.2l163.6 282.9H339zm-137.5 23.6L120.9 480h310.5L512 338.5H201.5zM154.1 67.4L0 338.5 80.6 480 237 208.8 154.1 67.4z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.twitter.com/your_twitter_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      className="p-4 col-md-10"
    >
      <form
        className="boxshadowAppoint p-4"
        onSubmit={[Function]}
      >
        <div>
          <label
            className="form-label"
            htmlFor="patientName"
          >
            Patient Name
          </label>
          <input
            className="form-control"
            id="patientName"
            name="patientName"
            readOnly={true}
            type="text"
            value={null}
          />
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="doctorId"
          >
            Select Doctor
          </label>
          <select
            className="form-control"
            id="doctorId"
            name="doctor_id"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select a doctor...
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="appointmentDate"
          >
            Appointment Date
          </label>
          <input
            className="form-control"
            id="appointmentDate"
            name="appointment_date"
            onChange={[Function]}
            required={true}
            type="datetime-local"
            value=""
          />
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.street"
          >
            Street
          </label>
          <select
            className="form-control"
            id="location.street"
            name="location.street"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select street...
            </option>
            <option
              value="524, Bhidan street"
            >
              524, Bhidan street
            </option>
            <option
              value="7895, Molla Hata street"
            >
              7895, Molla Hata street
            </option>
            <option
              value="9863, Dulhan street"
            >
              9863, Dulhan street
            </option>
            <option
              value="1545, Kabiry street"
            >
              1545, Kabiry street
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.state"
          >
            State
          </label>
          <select
            className="form-control"
            id="location.state"
            name="location.state"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select state...
            </option>
            <option
              value="Mymensignh"
            >
              Mymensignh
            </option>
            <option
              value="RBT"
            >
              RBT
            </option>
            <option
              value="NYC"
            >
              NYC
            </option>
            <option
              value="KTH"
            >
              KTH
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.city"
          >
            City
          </label>
          <select
            className="form-control"
            id="location.city"
            name="location.city"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select city...
            </option>
            <option
              value="Dhaka"
            >
              Dhaka
            </option>
            <option
              value="Chattagram"
            >
              Chattagram
            </option>
            <option
              value="Bhola"
            >
              Bhola
            </option>
            <option
              value="Jamalpur"
            >
              Jamalpur
            </option>
          </select>
        </div>
        <button
          className="mt-2 w-100 btn btn-primary"
          disabled={false}
          type="submit"
        >
          Add Appointment
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="appointmentBg row"
  >
    <div
      className="px-0 col-md-2"
    >
      <div
        className="d-flex flex-column border-end h-100 bg-white"
      >
        <div
          className="d-flex align-items-center p-3"
        >
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240.1 4.2c9.8-5.6 21.9-5.6 31.8 0l171.8 98.1L448 104l0 .9 47.9 27.4c12.6 7.2 18.8 22 15.1 36s-16.4 23.8-30.9 23.8H32c-14.5 0-27.2-9.8-30.9-23.8s2.5-28.8 15.1-36L64 104.9V104l4.4-1.6L240.1 4.2zM64 224h64V416h40V224h64V416h48V224h64V416h40V224h64V420.3c.6 .3 1.2 .7 1.8 1.1l48 32c11.7 7.8 17 22.4 12.9 35.9S494.1 512 480 512H32c-14.1 0-26.5-9.2-30.6-22.7s1.1-28.1 12.9-35.9l48-32c.6-.4 1.2-.7 1.8-1.1V224z"
              />
            </svg>
          </span>
          <h1
            className="m-0 ms-2"
          >
            Squa
          </h1>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M333.2 322.8l0 0-133.9-146 0 0L146 118.6c7.8-5.1 37-22.6 78-22.6s70.2 17.4 78 22.6L245.7 180l85.6 93.4 27.4-29.8c16.3-17.7 25.3-40.9 25.3-65V149.1c0-19-5.6-37.5-16.1-53.3L327.8 35.6C312.9 13.4 287.9 0 261.2 0h-76c-25.8 0-50.1 12.5-65.1 33.5L81.9 87C70.3 103.2 64 122.8 64 142.8V164c0 23.2 8.4 45.6 23.6 63.1l56 64.2 0 0 83.3 95.6 0 0 91.8 105.3c10 11.5 26.8 14.3 40 6.8l54.5-31.1c17.8-10.2 21.6-34.3 7.7-49.4l-87.7-95.7zM205.2 410.6l-83.3-95.6L27.1 418.5c-13.9 15.1-10.1 39.2 7.7 49.4l55.1 31.5c13 7.4 29.3 4.9 39.4-6.1l75.9-82.6z"
              />
            </svg>
          </span>
        </div>
        <div
          className="d-flex flex-column flex-grow-1 px-0"
        >
          <button
            className=" p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Add Doctor
          </button>
          <button
            className="p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={true}
            type="button"
          >
            Delete Doctor
          </button>
          <a
            activeClassName="active"
            className="p-3"
            href="/doctors"
            isActive={[Function]}
            onClick={[Function]}
          >
            Doctors
          </a>
          <a
            activeClassName="active"
            className="p-3"
            href="/appointment"
            isActive={[Function]}
            onClick={[Function]}
          >
            Appointment
          </a>
          <a
            activeClassName="active"
            className="p-3"
            href="/my_appointments"
            isActive={[Function]}
            onClick={[Function]}
          >
            My Appointments
          </a>
          <button
            className="p-3 dropdown-item"
            data-rr-ui-dropdown-item=""
            onClick={[Function]}
          >
            Sign Out
          </button>
          <button
            className=" p-3 add_doctor_btn nav-link  btn btn-primary"
            disabled={false}
            type="button"
          >
            For Super Admin
          </button>
        </div>
        <div
          className="d-flex align-items-center justify-content-between p-3 social_media"
        >
          <a
            className="nav-link"
            disabled={false}
            href="https://www.facebook.com/your_facebook_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.google.com/your_google_plus_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.linkedin.com/your_linkedin_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.google.com/your_google_drive_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M339 314.9L175.4 32h161.2l163.6 282.9H339zm-137.5 23.6L120.9 480h310.5L512 338.5H201.5zM154.1 67.4L0 338.5 80.6 480 237 208.8 154.1 67.4z"
              />
            </svg>
          </a>
          <a
            className="nav-link"
            disabled={false}
            href="https://www.twitter.com/your_twitter_profile_url"
            onClick={[Function]}
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      className="p-4 col-md-10"
    >
      <form
        className="boxshadowAppoint p-4"
        onSubmit={[Function]}
      >
        <div>
          <label
            className="form-label"
            htmlFor="patientName"
          >
            Patient Name
          </label>
          <input
            className="form-control"
            id="patientName"
            name="patientName"
            readOnly={true}
            type="text"
            value={null}
          />
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="doctorId"
          >
            Select Doctor
          </label>
          <select
            className="form-control"
            id="doctorId"
            name="doctor_id"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select a doctor...
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="appointmentDate"
          >
            Appointment Date
          </label>
          <input
            className="form-control"
            id="appointmentDate"
            name="appointment_date"
            onChange={[Function]}
            required={true}
            type="datetime-local"
            value=""
          />
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.street"
          >
            Street
          </label>
          <select
            className="form-control"
            id="location.street"
            name="location.street"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select street...
            </option>
            <option
              value="524, Bhidan street"
            >
              524, Bhidan street
            </option>
            <option
              value="7895, Molla Hata street"
            >
              7895, Molla Hata street
            </option>
            <option
              value="9863, Dulhan street"
            >
              9863, Dulhan street
            </option>
            <option
              value="1545, Kabiry street"
            >
              1545, Kabiry street
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.state"
          >
            State
          </label>
          <select
            className="form-control"
            id="location.state"
            name="location.state"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select state...
            </option>
            <option
              value="Mymensignh"
            >
              Mymensignh
            </option>
            <option
              value="RBT"
            >
              RBT
            </option>
            <option
              value="NYC"
            >
              NYC
            </option>
            <option
              value="KTH"
            >
              KTH
            </option>
          </select>
        </div>
        <div>
          <label
            className="form-label"
            htmlFor="location.city"
          >
            City
          </label>
          <select
            className="form-control"
            id="location.city"
            name="location.city"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              Select city...
            </option>
            <option
              value="Dhaka"
            >
              Dhaka
            </option>
            <option
              value="Chattagram"
            >
              Chattagram
            </option>
            <option
              value="Bhola"
            >
              Bhola
            </option>
            <option
              value="Jamalpur"
            >
              Jamalpur
            </option>
          </select>
        </div>
        <button
          className="mt-2 w-100 btn btn-primary"
          disabled={false}
          type="submit"
        >
          Add Appointment
        </button>
      </form>
    </div>
  </div>
</div>
`;
